integrationTests/chainSimulator/staking/stakeAndUnStake_test.go

```diff
@@ -0,0 +1,279 @@
+package staking
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	coreAPI "github.com/multiversx/mx-chain-core-go/data/api"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/integrationTests/chainSimulator/helpers"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/configs"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/dtos"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	defaultPathToInitialConfig             = "../../../cmd/node/config/"
+	maxNumOfBlockToGenerateWhenExecutingTx = 7
+)
+
+var log = logger.GetOrCreate("integrationTests/chainSimulator")
+
+// TODO scenarios
+// Make a staking provider with max num of nodes
+// DO a merge transaction
+
+// Test scenario
+// 1. Add a new validator private key in the multi key handler
+// 2. Do a stake transaction for the validator key
+// 3. Do an unstake transaction (to make a place for the new validator)
+// 4. Check if the new validator has generated rewards
+func TestChainSimulator_AddValidatorKey(t *testing.T) {
```

Comment:
suggestion: should we rename the scenarios to include the number from our internal testing plan for better traceability and documentation? cc @mariusmihaic @gabi-vuls @schimih